### PR TITLE
[SPARK-11894][SQL] fix isNull for GetInternalRowField

### PR DESCRIPTION
**Pull request by cloud-fan**
We should use `InternalRow.isNullAt` to check if the field is null before calling `InternalRow.getXXX`

Thanks @gatorsmile who discovered this bug.

### CODE REVIEW

**Comment by gatorsmile** [comment]
That is great! You fixed the bug! Thank you very much!

**Comment by cloud-fan** [comment]
cc @marmbrus

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala [hunk 1]
```diff
@@ -531,15 +526,15 @@ case class GetInternalRowField(child: Expression, ordinal: Int, dataType: DataTy
...
-    val row = child.gen(ctx)
-    s"""
-      ${row.code}
-      final boolean ${ev.isNull} = ${row.isNull};
-      ${ctx.javaType(dataType)} ${ev.value} = ${ctx.defaultValue(dataType)};
-      if (!${ev.isNull}) {
-        ${ev.value} = ${ctx.getValue(row.value, dataType, ordinal.toString)};
-      }
-    """
+    nullSafeCodeGen(ctx, ev, eval => {
```
Comment (marmbrus) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala]: This is fine for now, but this why we shouldn't have two different expressions for performing the same task.  We should have a single way to extract a field from a struct.

**Comment by marmbrus** [comment]
Thanks, merging to master and 1.6

**Pull request closed without merging**